source/Calamari/Deployment/Conventions/AzureWebAppConvention.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Web.Deployment;
+using Microsoft.WindowsAzure;
+using Microsoft.WindowsAzure.Management.WebSites.Models;
+using Octostache;
+
+namespace Calamari.Deployment.Conventions
+{
+    public class AzureWebAppConvention : IInstallConvention
+    {
+        readonly VariableDictionary variables;
+
+        public AzureWebAppConvention(VariableDictionary variables)
+        {
+            this.variables = variables;
+        }
+
+        public void Install(RunningDeployment deployment)
+        {
+            var subscriptionId = variables.Get(SpecialVariables.Machine.Azure.SubscriptionId);
+            var certificate = Convert.FromBase64String(variables.Get(SpecialVariables.Machine.Azure.CertificateBytes));
+            var webAppName = variables.Get(SpecialVariables.Machine.Azure.WebAppName);
+            var webSpaceName = variables.Get(SpecialVariables.Machine.Azure.WebSpaceName);
+
+            Log.Info("Deploying to Azure WebApp {0} in WebSpace {1} using subscription-id {2}", webAppName, webSpaceName, subscriptionId);
+
+            var cloudClient = CloudContext.Clients.CreateWebSiteManagementClient( 
```

Comment:
I intend to factor some of this work out of the convention once we implement the next azure-target (e.g. cloud-services).